extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -259,31 +263,61 @@ export async function runRemoteQuery(
     });
 
     const actionBranch = getActionBranch();
-    const apiResponse = await runRemoteQueriesApiRequest(credentials, actionBranch, language, repoSelection, owner, repo, base64Pack, dryRun);
     const queryStartTime = Date.now();
     const queryMetadata = await tryGetQueryMetadata(cliServer, queryFile);
 
-    if (dryRun) {
-      return { queryDirPath: remoteQueryDir.path };
+    if (isVariantAnalysisLiveResultsEnabled()) {
+      const queryName = getQueryName(queryMetadata, queryFile);
+      const variantAnalysisSubmission: VariantAnalysisSubmission = {
+        startTime: queryStartTime,
+        actionRepoRef: actionBranch,
+        controllerRepoId: controllerRepoId,
+        query: {
+          name: queryName,
+          filePath: queryFile,
+          pack: base64Pack,
+          language: language as VariantAnalysisQueryLanguage,
```

Comment:
Since this is being cast to the `VariantAnalysisQueryLanguage`, should we validate that the `language` is a supported MRVA language?

Reply:
I'm not sure. On the one hand it's nice to have accurate types, but then it's seems weird to do the type checking here. We are storing the language as a string elsewhere in the extension, so shouldn't we be applying this type earlier on (I realise we've only just introduced the type which is why it's not used elsewhere).

On the other hand the API itself does validation that the language is from the known set of languages, and returns an error message. So just passing the string to the API and letting it decide if the language is ok or not also seems like a good option to me.

Another question is are we going to ever do anything in the VSCode extension that depends on the exact language? I can't think of many cases. We'll generally either just display the language or pass it to other places like the CodeQL CLI or the github API. We might want to display a better formatted name (i.e. "JavaScript" instead of "javascript") but that's about all.

So my preference would actually be to delete the `VariantAnalysisQueryLanguage` type and just treat the language as a string. We could also keep the type but make it an alias of string. I just don't know if that's too radical of a decision to make.

What do you think?

Reply:
I'm changing my mind. For this PR I'll just add a method to check if the language is valid or not, and we'll throw an exception if it's not. I'll leave the discussion of whether we want this type of not for a later time.